### ISSUE TITLE
`torch.{max,min}` have strange error message when `input.numel()==0`

### ISSUE THREAD

**Issue by TestSomething22** [pytorch/pytorch#71629]
### 🐛 Describe the bug

`torch.{max,min}` have strange error message when `input.numel()==0`. It says "Expected reduction dim to be specified for input.numel() == 0. Specify the reduction dim with the 'dim' argument". However, it is meaningless to specify the `dim` for an empty tensor. By contrast, `torch.median` will return `tensor(nan)` when the input is empty

```python
import torch
input = torch.rand([5, 5, 0])
torch.median(input)
# tensor(nan)
torch.max(input)
# ERROR:max(): Expected reduction dim to be specified for input.numel() == 0. Specify the reduction dim with the 'dim' argument.
```

### Versions

pytorch: 1.10.1

**Comment by flashasdbaksdgi** [comment]
Hi 
did u sovle this problem yet?
how?

**Comment by TestSomething22** [comment]
Hi @flashasdbaksdgi, I do not how to deal with this problem so let me reopen it.
